### PR TITLE
Admin API and Admin GUI

### LINKED ISSUE #26: Split Admin Rest into api and web

### LINKED ISSUE #27: Remove the conn strings from clear text

### PR DESCRIPTION
**Pull request by rfrsilva**
Closes #26 
Closes #27 

- Split Admin into an API and a GUI;
- Hash passwords using Argon2;
- Remove plain-text passwords of connection strings;
- New design of Create Probe page;
- New Error page.

### CODE REVIEW

File: development/tma-admin-api/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+RUN     ["mvn", "dependency:resolve"]
+RUN     ["mvn", "verify"]
+
+# Adding source, compile and package into a fat jar
+ADD     src      ${adminapi}/tma-admin-api/src
+RUN     ["mvn", "clean", "install"]
+
+RUN     ["cp", "-r", "bin", "/atmosphere/tma/admin-api/bin"]
+
+CMD ["java", "-jar", "/atmosphere/tma/admin-api/bin/tma-admin-rest-0.0.1-SNAPSHOT.jar"]
```
Comment (nmsa) [inline development/tma-admin-api/Dockerfile]: It is important to understand that using this path, the running path is not the one where the jar is located, but instead the current WORKDIR (see line 7).
Is this relevant?

File: development/tma-admin-web/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+RUN     ["mvn", "dependency:resolve"]
+RUN     ["mvn", "verify"]
+
+# Adding source, compile and package into a fat jar
+ADD     src      ${adminweb}/tma-admin-web/src
+RUN     ["mvn", "clean", "install"]
+
+RUN     ["cp", "-r", "bin", "/atmosphere/tma/admin-web/bin"]
+
+CMD ["java", "-jar", "/atmosphere/tma/admin-web/bin/tma-admin-web-0.0.1-SNAPSHOT.jar"]
```
Comment (nmsa) [inline development/tma-admin-web/Dockerfile]: Same comment as above.
The main point where it may be relevent is in the logs.

**Comment by nmsa** [comment]
@pgoncalvesgit @rfrsilva please, check if all the dependencies of `tma-admin-web` are really necessary.

File: development/tma-admin-api/Dockerfile (hunk 1 above)
Comment (rfrsilva) [reply development/tma-admin-api/Dockerfile]: Done

File: development/tma-admin-web/Dockerfile (hunk 2 above)
Comment (rfrsilva) [reply development/tma-admin-web/Dockerfile]: Done

**Comment by rfrsilva** [comment]
Only log4j dependency is needed.
Unnecessary dependencies removed in this [commit](https://github.com/eubr-atmosphere/tma-framework-k/pull/31/commits/bb8d1ee3c4df178edb6342c0535a89ad89d07772).

File: development/tma-admin-api/src/main/resources/application.properties [hunk 3]
```diff
@@ -1,6 +1,8 @@
 partnerList=UPV:10,UFCG:15,TRUST-IT SERVICES:20,UFMG:25,UC:30,UNICAMP:35,POLIMI:40,UFAM:45,TUD:50,UnB:55,UPRC:60,KUNUMI:65,QUIMBIM:70,DELL-EMC:80
 algorithm=RSA
 userDB=root
+maxNumbeOfDatabaseConnections=1
```
Comment (joseadp) [inline development/tma-admin-api/src/main/resources/application.properties]: typo

File: development/tma-admin-api/src/main/java/eu/atmosphere/tma/admin/util/DatabaseManager.java [hunk 4]
```diff
@@ -734,13 +838,15 @@ public boolean deleteConfiguration(int configurationId) {
...
-        Connection con = getConnection();
+    public ArrayList<Probe> getProbes() throws SQLException {
+        
+        Connection con = DatabaseManager.getConnectionFromPool();
         try {
             ArrayList<Probe> probes = new ArrayList<>();
             try (Statement statement = con.createStatement()) {
                 ResultSet resultSet = statement.executeQuery(SQL_SELECT_PROBE);
                 while (resultSet.next()) {
+                    System.out.println(resultSet.getTimestamp("tokenExpiration"));
```
Comment (joseadp) [inline development/tma-admin-api/src/main/java/eu/atmosphere/tma/admin/util/DatabaseManager.java]: Change to LOGGER if it is needed.

File: development/tma-admin-api/src/main/java/eu/atmosphere/tma/admin/util/DatabaseManager.java [hunk 5]
```diff
@@ -53,7 +54,35 @@
...
+                PropertiesManager.getInstance().getProperty("passwordProduction"));
+        String password = new String(decoded);
+        CONFIG.setUsername(userDB);
+        CONFIG.setPassword(password);
+        CONFIG.addDataSourceProperty("cachePrepStmts", "true");
+        CONFIG.addDataSourceProperty("prepStmtCacheSize", "250");
+        CONFIG.addDataSourceProperty("prepStmtCacheSqlLimit", "2048");
+        CONFIG.setAutoCommit(false);
+        CONFIG.setMaximumPoolSize(Integer.parseInt(
+                PropertiesManager.getInstance().getProperty("maxNumbeOfDatabaseConnections")));
```
Comment (joseadp) [inline development/tma-admin-api/src/main/java/eu/atmosphere/tma/admin/util/DatabaseManager.java]: typo

**Pull request merged**